ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+          towerColor = data.engravement3TowerType === 'lightTower'
+            ? output.light!()
+            : output.dark!();
+        let partner;
+        if (towerOnYou) {
+          partner = data.ShortName(data.engravement3TowerPlayers.find((name) =>
+            name !== data.me
+          )) ?? output.unknown!();
+          return { alertText: output.towerOnYou!({ color: towerColor, partner: partner }) };
+        } else if (data.party.isDPS(data.me)) {
+          // DPS want to know tower color to know if they are soaking or baiting add cleaves later
+          return { infoText: output.towerLater!({ color: towerColor }) };
+        }
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Towers Drop',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTowerIds },
+      condition: (data, matches) => data.engravementCounter === 3 && data.me === matches.target,
+      delaySeconds: (_data, matches) => parseFloat(matches.duration) - 5,
+      alertText: (data, _matches, output) => {
+        let towerColor = output.unknown!();
+        if (data.engravement3TowerType !== undefined)
+          towerColor = data.engravement3TowerType === 'lightTower'
+            ? output.light!()
+            : output.dark!();
+        return output.dropTower!({ color: towerColor });
+      },
+      outputStrings: {
+        dropTower: {
+          en: 'Drop ${color} Tower',
+        },
+        light: {
+          en: 'Light',
+        },
+        dark: {
+          en: 'Dark',
+        },
+        unknown: Outputs.unknown,
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Soak Tower/Bait Adds',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTiltIds },
+      condition: (data, matches) => data.engravementCounter === 3 && data.me === matches.target,
+      suppressSeconds: 15, // avoid second (incorrect) alert when debuff switches from soaking tower
+      alertText: (data, matches, output) => {
+        // lightTower/darkTower support players receive lightTilt/darkTilt once dropping their tower
+        // so exclude them from receiving this alert
+        if (data.engravement3TowerPlayers.indexOf(data.me) !== -1)
+          return;
+
+        const soakMap: { [key: string]: string } = {
+          lightTower: 'darkTilt',
+          darkTower: 'lightTilt',
+        };
+        const myEffect = engravementLabelMap[matches.effectId];
+        if (myEffect === undefined || data.engravement3TowerType === undefined)
+          return;
+        const soakers = soakMap[data.engravement3TowerType];
+        if (soakers === undefined)
+          return;
+
+        const towerColor = data.engravement3TowerType === 'lightTower'
+          ? output.light!()
+          : output.dark!();
+        if (myEffect === soakers)
+          return output.soakTower!({ color: towerColor });
+        return output.baitCleaves!();
+      },
+      outputStrings: {
+        soakTower: {
+          en: 'Soak ${color} Tower',
```

Comment:
Unless somebody comes up with some wild uptime strategy that puts people in different spots, would it be possible to tell the dps soaking here to soak the middle tower vs soaking the "non-middle" (idk what to call this) tower as well?

Reply:
> Unless somebody comes up with some wild uptime strategy that puts people in different spots, would it be possible to tell the dps soaking here to soak the middle tower vs soaking the "non-middle" (idk what to call this) tower as well?

I don't think we can tell the DPS which tower to soak, because both DPS are on the same platform, and one towerr will be dropped on the platform and one will be dropped by the support player on the other platform (but on the intersecting corner).  But either DPS can soak either tower, and I've seen both strats (having the diagonal tether soak the center one, or the straight tether soaking the center one), not to mention the 'yolo' strat (which seems to happen 90% of the time anyway).

I also thought about whether we could remind the *support* player in 'P12S Engravement 3 Towers Drop' to drop their tower center/overlapping if both DPS on their side have the same color tether.  And while we can tell whether the East or West support player will need to drop mid (based on the positioning of the adds and their tether colors), we would have to base that call on whether the support player is standing east or west at the time (and since we know the center point, player position (if accurate) should be a 100% reliable way to determine that).

Does `getCombatants` return player data as well, or just NPC data?

Reply:
Ohhhhhhh that's right.  Sorry, I was thinking about this wrong.  They're either soaking towers or baiting, so you can't tell them which.  Yeah, I think the support person should get the call about where to place the tower, and the dps can just sort it out.

We should be able to know which side they're on based on player position from the taking the tower damage if you want a non-`getCombatants` method.

Reply:
> We should be able to know which side they're on based on player position from the taking the tower damage if you want a non-`getCombatants` method.

Brill.  Working on it.